ddtrace/contrib/psycopg/async_cursor.py

```diff
@@ -21,6 +26,33 @@ async def __aexit__(self, exc_type, exc_val, exc_tb) -> None:
         # messages will be the same.
         return await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
 
+    def __aiter__(self):
```

Comment:
Does this follow the behavior of the original Pyscopg3AsyncCursor? I think that this probably avoids the error, but would lead to unexpected behavior since it seems that this just returns self, and the expected effect of aiter is: https://github.com/psycopg/psycopg/blob/9c8664117ff20fff7eb3e72c31868764439eddcd/psycopg/psycopg/cursor_async.py#L239-L251

Reply:
I was referencing other places in the repo where we implement asynchronous iteration via the `_aiter_` and `_anext_` methods: https://github.com/DataDog/dd-trace-py/blob/3237351b4a35dfd52e3d3d05c81bfe5b51279372/ddtrace/contrib/openai/utils.py#L106
so in the case of processing rows, it would just call the `_anext_` method until `row is None` which raises `StopAsyncIteration`.

However, reading [here](https://peps.python.org/pep-0525/) it seems like async generators are more performant than async iterators anyway, which would especially matter in the case of handling large datasets. So I can switch back to the original implementation.